File: src/generate_random_walk_ts.py
# Generate random data

import numpy as np
import matplotlib.pyplot as plt
from statsmodels.tsa.arima_process import arma_generate_sample
np.random.seed(42)

# Coefficients
ar_coefs = [1]
ma_coefs = [1]

# Generate ARMA data
y = arma_generate_sample(ar_coefs, ma_coefs, nsample = 200, scale = .5)
y = np.cumsum(y) + 20

# Plot random walk time series
plt.clf()
plt.plot(y)
plt.ylabel(r'$y_t$')
plt.xlabel(r'$t$')
plt.title("Random Walk Time Series")
plt.show()

File: src/generate_random_walk_ts.R
# Generate random data

library(ggplot2)
library(tidyverse)

set.seed(42)

# Generate ARMA data
y <- arima.sim(model = list(order = c(0, 1, 0)),
               n = 200,
               sd = 0.5) + 20

m <- length(y)
y_df <- data.frame(time = 1:m, y =  y)

# Plot random walk time series
y_df %>%
  ggplot(aes(x = time, y = y)) +
  geom_line() +
  labs(
    title = "Random Walk Time Series",
    x = "Time",
    y = "y"
  ) +
  scale_y_continuous() +
  theme_minimal()
